Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -443,13 +462,37 @@ - (void) receivedProxyHTTPResponse: (CFHTTPMessageRef)httpResponse {
 - (void) receivedHTTPResponse: (CFHTTPMessageRef)httpResponse {
     // Post the response headers to LiteCore:
     NSDictionary *headers =  CFBridgingRelease(CFHTTPMessageCopyAllHeaderFields(httpResponse));
+    
+    // save to lite core
     NSString* cookie = headers[@"Set-Cookie"];
-    if ([cookie containsString: @", "]) {
-        // CFHTTPMessage incorrectly merges multiple Set-Cookie headers. Undo that:
-        NSMutableDictionary* newHeaders = [headers mutableCopy];
-        newHeaders[@"Set-Cookie"] = [cookie componentsSeparatedByString: @", "];
-        headers = newHeaders;
+    if (cookie.length > 0) {
+        C4Error err = {};
+        if (!c4db_setCookie(_c4db, cookie.c4slice, _remoteURL.host.c4slice,
+                            _remoteURL.path.stringByDeletingLastPathComponent.c4slice, &err)) {
+            CBLWarnError(WebSocket, @"Cannot save cookie %d/%d", err.domain, err.code);
+            // TODO: any more handling?
+        }
+        
+        // update the response headers! do we really need this now? since we are already saving to c4db?
```

Comment:
We still need to break the merged set-cookies and I don't know if LiteCore still handles anything related the cookies. I think it's no harm to send the correct header to LiteCore.